library/net_sockets.c

```diff
@@ -270,13 +270,18 @@ static int net_would_block( const mbedtls_net_context *ctx )
  */
 static int net_would_block( const mbedtls_net_context *ctx )
 {
+    int err = errno;
+    
     /*
      * Never return 'WOULD BLOCK' on a non-blocking socket
      */
     if( ( fcntl( ctx->fd, F_GETFL ) & O_NONBLOCK ) != O_NONBLOCK )
```

Comment:
If `fcntl` fails then it returns -1. Assuming two's complement (mandated by POSIX, and so extremely likely to be the case on any platform where this code compile, not that non-two's-complement platforms are common), this is all-bit-one and so the code goes on to check the previous value of `errno`.

The only documented failure for `fcntl(fd, F_GETFL)` is an invalid file descriptor. I think in that case it would be best to return 0, which should make the caller fail early as it attempts whatever it was thinking of doing.